clio-client/src/test/scala/org/broadinstitute/clio/client/dispatch/AmendExecutorSpec.scala

```diff
@@ -0,0 +1,174 @@
+package org.broadinstitute.clio.client.dispatch
+
+import java.net.URI
+
+import akka.stream.scaladsl.{Sink, Source}
+import io.circe.syntax._
+import org.broadinstitute.clio.client.BaseClientSpec
+import org.broadinstitute.clio.client.commands.AmendArrays
+import org.broadinstitute.clio.client.util.IoUtil
+import org.broadinstitute.clio.client.webclient.ClioWebClient
+import org.broadinstitute.clio.client.webclient.ClioWebClient.{QueryAux, UpsertAux}
+import org.broadinstitute.clio.transfer.model.ArraysIndex
+import org.broadinstitute.clio.transfer.model.arrays.{
+  ArraysKey,
+  ArraysMetadata,
+  ArraysQueryInput
+}
+import org.broadinstitute.clio.transfer.model.gvcf.GvcfMetadata
+import org.broadinstitute.clio.util.model.{Location, UpsertId}
+import org.scalamock.scalatest.AsyncMockFactory
+
+import scala.collection.immutable
+
+class AmendExecutorSpec extends BaseClientSpec with AsyncMockFactory {
+  behavior of "AmendExecutor"
+
+  private val documentsInClio = 0 until 2000
+  private val alreadyHasMetadata = 500
+
+  private val loc = URI.create("some/fake/metadata.json")
+  private val keys = documentsInClio map { i =>
+    ArraysKey(Location.GCP, Symbol(s"chipwell_barcode_$i"), i)
+  }
+  private val metadatas = documentsInClio map { i =>
+    ArraysMetadata(
+      gtcPath = Some(URI.create(s"gs://some-bucket/some_path/v$i/the_gtc.gtc"))
+    )
+  }
+
+  private val zipped = keys.zip(metadatas)
+
+  val newMetadata = ArraysMetadata(sampleAlias = Some("amended_sample_alias"))
+
+  private def setupQueryReturn(
+    webClient: ClioWebClient,
+    documents: immutable.Iterable[(ArraysKey, ArraysMetadata)]
+  ): Unit = {
+    val _ = (
+      webClient
+        .simpleQuery(_: QueryAux[ArraysQueryInput])(
+          _: ArraysQueryInput,
+          _: Boolean
+        )
+      )
+      .expects(ArraysIndex, *, *)
+      .returning(Source(documents.map { km =>
+        km._1.asJson.deepMerge(km._2.asJson)
+      }))
+  }
+
+  private def setupUpserts(
+    webClient: ClioWebClient,
+    documents: immutable.Iterable[(ArraysKey, ArraysMetadata)]
+  ): Unit = {
+
+    documents.foreach { km =>
+      val (theKey, metadata) = km
+      (
+        webClient
+          .upsert(_: UpsertAux[ArraysKey, ArraysMetadata])(
+            _: ArraysKey,
+            _: ArraysMetadata,
+            _: Boolean
+          )
+        )
+        .expects(ArraysIndex, theKey, metadata, false)
+        .returning(Source.single(UpsertId.nextId().asJson))
+    }
+  }
+
+  it should "not amend documents that already have metadata defined" in {
+    val ioUtil = mock[IoUtil]
+    (ioUtil.readFileData _)
+      .expects(loc)
+      .returning(newMetadata.asJson.pretty(defaultPrinter))
+
+    val webClient = mock[ClioWebClient]
```

Comment:
If you have the spec extend OneInstancePerTest, you can share a single mock between the tests without worrying about expectations from one test interfering with another.